Project_One/app/src/main/java/com/justinwells/project_one/MainActivity.java

```diff
@@ -0,0 +1,118 @@
+    List <String> title;
+    ArrayAdapter<String> arrayAdapter;
+    public static int workingIndex = 0;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        title = new ArrayList<>();
+        listView = (ListView) findViewById(R.id.list_view);
+        arrayAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1,title);
+        listView.setAdapter(arrayAdapter);
+
+        final Intent intent = new Intent(MainActivity.this, Main2Activity.class);
+
+
+        FloatingActionButton addList = (FloatingActionButton) findViewById(R.id.button);
+
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, final int position, long id) {
+
+                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+
+                builder.setTitle("What do you want to do?");
+
+
+                builder.setPositiveButton("Remove List", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        ListManager listManager = ListManager.getInstance();
+                        listManager.removeList(position);
+                        title.remove(position);
+
+                        arrayAdapter.notifyDataSetChanged();
+                    }
+                });
+
+                builder.setNegativeButton("View List", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        workingIndex = position;
+                        startActivityForResult(intent, LIST_PAGE);
+                    }
+                });
+
+                AlertDialog dialog = builder.create();
+                dialog.show();
+
+            }
+        });
+
+        addList.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+
+                builder.setTitle("Enter List Name");
+                final EditText name = new EditText(MainActivity.this);
+
+                name.setHint("My List");
+                builder.setView(name);
+
+
+                builder.setPositiveButton("CONFIRM", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        ListManager listManager = ListManager.getInstance();
+
+                        String listTitle = name.getText().toString();
+
+                        if (listTitle.equals("") || listTitle == null) {
+                            listTitle = "My list";
```

Comment:
Per the requirements, you should be calling `name.setError()` here rather than specifying a default list name. I suspect you did this because the alert dialog automatically closes when you set the error message, so the user never sees it. There are a few ways around this; the simplest is to first show the dialog, then set the listener for the positive button: http://stackoverflow.com/a/6142413